superset/db_engine_specs/pinot.py

```diff
@@ -44,13 +45,32 @@ def get_timestamp_expr(
         cls, col: ColumnClause, pdf: Optional[str], time_grain: Optional[str]
     ) -> TimestampExpression:
         is_epoch = pdf in ("epoch_s", "epoch_ms")
-        if not is_epoch:
-            raise NotImplementedError("Pinot currently only supports epochs")
+
         # The DATETIMECONVERT pinot udf is documented at
         # Per https://github.com/apache/incubator-pinot/wiki/dateTimeConvert-UDF
         # We are not really converting any time units, just bucketing them.
-        seconds_or_ms = "MILLISECONDS" if pdf == "epoch_ms" else "SECONDS"
-        tf = f"1:{seconds_or_ms}:EPOCH"
+        tf = ""
+        if not is_epoch:
+            try:
+                today = datetime.datetime.today()
+                today.strftime(str(pdf))
+            except ValueError:
+                raise NotImplementedError(
+                    "Pinot currently doesn't support date formate: " + str(pdf)
```

Comment:
Typo (formate -> format) + please use f-string. Also, the error message indicates that the error is specific to Pinot; it is my understanding that `ValueError` in this case will be thrown if it is not a valid python date format.

Reply:
You are correct, now we support datetime format, so will raise a ValueError here.